src/parser.rs

```diff
@@ -229,6 +243,24 @@ named!(gosub_statement<CompleteStr, Statement>,
         (Statement::Gosub(line_number))
     ));
 
+// TODO: Restrict to numericexpr-relation-numericexpr
+// or stringexpr-relation-stringexpr
+named!(if_then_statement<CompleteStr, Statement>,
+    do_parse!(
```

Comment:
I think you can do an alt! tag here, and use `string_expression` or `numeric_expression` to restrict resolve your TODO.

Reply:
I fail on runtime, which adheres slightly better to the nature of BASIC eheh

Reply:
But then this is not according to the spec, which explicetly sais that this should not be possible.

Reply:
Does the spec separate the parsing and interpreting part?

Reply:
It says: `relational-expression = numeric-expression relation numeric-expression | string-expression relation string-expression`. Right now, we have `relational-expression = expression relation expression`.